Resources/public/js/views/universaldiscovery/ez-universaldiscoverysearchview.js

```diff
@@ -0,0 +1,500 @@
+            this.after('visibleChange', this._unselectContent);
+        },
+
+        /**
+         * Custom reset implementation to explicitely reset the sub views.
+         *
+         * @method reset
+         * @param {String} [name]
+         */
+        reset: function (name) {
+            if ( name === 'selectedView' ) {
+                this.get('selectedView').reset();
+                return;
+            }
+            this.constructor.superclass.reset.apply(this, arguments);
+        },
+
+        render: function () {
+            var container = this.get('container');
+
+            container.setHTML(this.template({
+                searchText: this.get('searchText'),
+                searchResultCount: this.get('searchResultCount'),
+                searchResultList: this._convertToJSONList(),
+                isFirst: this._isFirstPage(),
+                isLast: this._isLastPage(),
+                hasPages: this._hasPages(),
+                loadingError: this.get('loadingError'),
+                multiple: this.get('multiple')
+            }));
+
+            this.get('selectedView').set('contentStruct', null);
+            container.one('.ez-ud-search-selected').append(
+                this.get('selectedView').render().get('container')
+            );
+            this._uiPageEndLoading();
+            return this;
+        },
+
+        onUnselectContent: function (contentId) {
+            var selectedViewStruct = this.get('selectedView').get('contentStruct');
+
+            if ( selectedViewStruct && selectedViewStruct.contentInfo.get('id') === contentId ) {
+                this.get('selectedView').set('confirmButtonEnabled', true);
+            }
+        },
+
+        /**
+         * `multipleChange` and `isSelectableChange` events handler. It sets the selected view
+         * `addConfirmButton` flag according to the new `multiple` attribute value and passes
+         * new `isSelectable` function to the selected view.
+         *
+         * @method _setSelectedViewAttrs
+         * @protected
+         */
+        _setSelectedViewAttrs: function () {
+            this.get('selectedView').setAttrs({
+                'addConfirmButton': this.get('multiple'),
+                'isSelectable': this.get('isSelectable')
+            });
+        },
+
+        /**
+         * Search button `tap` event handler. It sets the attributes that take part in the search
+         * and proceeds with the firing location search.
+         *
+         * @method _search
+         * @protected
+         * @param {EventFacade} e
+         */
+        _search: function (e) {
+            var c = this.get('container'),
+                searchInput = c.one(e.target.getAttribute('data-search-input-selector')),
+                searchText = searchInput.get('value');
+
+            e.preventDefault();
+
+            this.set('searchText', searchText);
+            this.set('offset', 0);
+            this._fireLocationSearch();
```

Comment:
I have removed that from `_search` method and subscribed it to `searchTextChange` event.